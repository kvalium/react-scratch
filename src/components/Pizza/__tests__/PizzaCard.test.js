import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import jest from 'jest-mock';

import PizzaCard from '../PizzaCard';

const fakePizza = {
  id: 1,
  name: 'Calzone',
  desc: 'fake pizza',
  price: 10,
};

/**
 * SNAPSHOT TEST
 */
test('renders', () => {
  const component = renderer.create(<PizzaCard pizza={fakePizza} onAddToCart={() => jest.fn()} />);
  expect(component.toJSON()).toMatchSnapshot();
});

/**
 * EVENT HANDLER TEST
 */
test('triggers onAddCart', () => {
  const addToCartMock = jest.fn();

  const wrapper = shallow(<PizzaCard pizza={fakePizza} onAddToCart={addToCartMock} />);
  wrapper.find('button').simulate('click');
  expect(addToCartMock).toHaveBeenCalledTimes(1);
});
