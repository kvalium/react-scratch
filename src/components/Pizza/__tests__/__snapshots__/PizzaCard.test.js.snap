// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="pizza-link"
>
  <div
    className="card"
  >
    <div
      className="card-content"
    >
      <div
        className="media"
      >
        <div
          className="media-content"
        >
          <div
            className="columns"
          >
            <div
              className="column is-9"
            >
              <p
                className="title is-4"
              >
                Calzone
              </p>
            </div>
            <div
              className="column is-3"
            >
              <p
                className="title is-4 has-text-primary"
              >
                10 â‚¬
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="content"
      >
        fake pizza
      </div>
      <button
        className="button is-primary is-small is-rounded"
        onClick={[Function]}
        type="button"
      >
        Ajouter au panier
      </button>
    </div>
  </div>
</div>
`;
